packages/airnode-adapter/src/response-processing/extraction.test.ts

```diff
@@ -66,11 +64,95 @@ describe('extractAndEncodeValue', () => {
   it('extracts and encodes the value from complex objects', () => {
     const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
     const parameters: ReservedParameters = { _path: 'a.b.1.d', _type: 'int256', _times: '100' };
-    const res = extraction.extractAndEncodeResponse(data, parameters);
+    const res = extractAndEncodeResponse(data, parameters);
     expect(res).toEqual({
-      rawValue: '750.51',
-      value: '75051',
+      rawValue: data,
+      values: ['75051'],
       encodedValue: '0x000000000000000000000000000000000000000000000000000000000001252b',
     });
   });
+
+  it('accepts "" (empty string) for _times parameter', () => {
+    const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
+    const parameters: ReservedParameters = { _path: 'a.b.1.d', _type: 'int256', _times: '' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: ['750'],
+      encodedValue: '0x00000000000000000000000000000000000000000000000000000000000002ee',
+    });
+  });
+
+  it('empty string in path returns the whole API response', () => {
+    const data = 123456;
+    const parameters: ReservedParameters = { _path: '', _type: 'int256' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: 123456,
+      values: ['123456'],
+      encodedValue: '0x000000000000000000000000000000000000000000000000000000000001e240',
+    });
+  });
+});
+
+describe('extract and encode multiple values', () => {
+  it('works for basic request', () => {
+    const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
+    const parameters: ReservedParameters = { _path: 'a.b.1.d,a.b.0.c', _type: 'int256,bool', _times: '100,' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
```

Comment:
I needed to change the signature of the encoding function. Airnode mostly cares about the `encodedValue` which meaning didn't change.

However, I've changed testing gateway output. @andreogle this maybe impacts chainAPI